cmd/snap/cmd_quota_test.go

```diff
@@ -181,14 +193,31 @@ func (s *quotaSuite) TestSetQuotaInvalidArgs(c *check.C) {
 		{[]string{"set-quota", "--memory=99B"}, "the required argument `<group-name>` was not provided"},
 		{[]string{"set-quota", "--memory=99", "foo"}, `cannot parse "99": need a number with a unit as input`},
 		{[]string{"set-quota", "--memory=888X", "foo"}, `cannot parse "888X\": try 'kB' or 'MB'`},
+		{[]string{"set-quota", "--cpu=0", "foo"}, `invalid cpu quota format specified for --cpu`},
+		{[]string{"set-quota", "--cpu=0x100", "foo"}, `invalid cpu quota format specified for --cpu`},
+		{[]string{"set-quota", "--cpu=200", "foo"}, `invalid cpu quota format specified for --cpu`},
+		{[]string{"set-quota", "--cpu=20D", "foo"}, `invalid cpu quota format specified for --cpu`},
+		{[]string{"set-quota", "--cpu=ASD", "foo"}, `invalid cpu quota format specified for --cpu`},
+		{[]string{"set-quota", "--cpu-set=x", "foo"}, `cannot parse value for --cpu-set at position 0`},
+		{[]string{"set-quota", "--cpu-set=1:2", "foo"}, `cannot parse value for --cpu-set at position 0`},
+		{[]string{"set-quota", "--cpu-set=60000", "foo"}, `invalid cpu number 60000 in --cpu-set`},
+		{[]string{"set-quota", "--cpu-set=0,-2", "foo"}, `invalid cpu number -2 in --cpu-set`},
 		// remove-quota command
 		{[]string{"remove-quota"}, "the required argument `<group-name>` was not provided"},
 	} {
 		s.stdout.Reset()
 		s.stderr.Reset()
 
+		// Mock the CGroup version here to test the --cpu-set command
+		restore := main.MockGetCGroupVersion(func() (int, error) {
```

Comment:
we probably need a test to verify that --cpu-set complains with v1, but --cpu works

Reply:
I have added test for this case, good catch